### PR TITLE
feature: all de/serializers support content type

### PR DESCRIPTION
**Pull request by athewsey**
**Issue #, if available:** #1977

**Description of changes:**

Add support for `content_type` constructor argument in all provided `Serializer`s, and for `accept` constructor argument in all provided `Deserializer`s.

This change will reduce the frequency with which users need to create custom de/serializers, by increasing the flexibility of the pre-provided ones.

For example: consider the built-in [SageMaker Semantic Segmentation algorithm](https://docs.aws.amazon.com/sagemaker/latest/dg/semantic-segmentation.html#semantic-segmentation-inputoutput), where `BytesDeserializer` may be appropriate for receiving an `image/png` result - but the `Accept` header must explicitly indicate that this result is preferred rather than an `application/x-recordio-protobuf`.

I've strived for maximal backward compatibility in the proposed changeset, and believe the change is non-breaking:

- The `SimpleBase*` classes are introduced as a compromise between wanting to keep code DRY / implementing the API exactly once, versus not wanting to change the advertised `Base*` APIs.
- The new arguments are always optional, and appended to existing lists where necessary in case anybody's been using existing kwargs by position.
- Although I don't really like the ambiguity of implementing constant-looking readonly props `ACCEPT` and `CONTENT_TYPE` backed by non-private-marked instance variables `accept` and `content_type` (rather than e.g. `_accept`, and maybe making these lowercase names the actual read/writable properties) - I kept the pattern for consistency with existing implementations in e.g. `NumpyDeserializer` and `IdentitySerializer`.

I guess one small doc change introduced is that the `NumpyDeserializer` in particular used to carry a warning that multiple accept types may not work with legacy images. That warning is no longer present and it's now possible to assign multiple content types to this Deserializer... Maybe could consider adding some generic wording to *every* `accept` docstring?

Also added a bit of a note on the `CSVDeserializer` docstring as I only realised while making this change that the `NumpyDeserializer` supports additional content-types and is a good choice for text/csv->ndarray!

**Testing done:**

Because the change is implemented via optional constructor args, no new code paths are introduced and so no new tests have been added.

I've checked the existing unit tests all pass, and have done some anecdotal experiments with a couple of classes to check they seem to work as expected.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] ~I have passed the region in to all S3 and STS clients that I've initialized as part of this change.~
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] ~I have added tests that prove my fix is effective or that my feature works (if appropriate)~
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] ~I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by athewsey** [comment]
^ Looking at the log, I think that CI job had some bigger problem than my changeset? But happy to look at fixing it if somebody disagrees!

**Comment by athewsey** [comment]
OK the second one (integration) looks like it was my bad as some tests are (incorrectly I might add!) trying to set `CONTENT_TYPE` - will get on that

File: src/sagemaker/deserializers.py [hunk 1]
```diff
@@ -55,17 +56,44 @@ def ACCEPT(self):
...
+        Args:
+            accept (union[str, tuple[str]]): The MIME type (or tuple of allowable MIME types) that
+                is expected from the inference endpoint (default: "*/*").
+        """
+        super(SimpleBaseDeserializer, self).__init__()
+        self.accept = accept
+
+    @property
+    def ACCEPT(self):
+        """The tuple of possible content types that are expected from the inference endpoint."""
```
Comment (icywang86rui) [inline src/sagemaker/deserializers.py:79]:
```suggestion
        """Content type or the tuple of possible content types that are expected from the inference endpoint.
            
            Returns: MIME type or tuple of MIME types.
        """
```

**Review by icywang86rui — COMMENTED** [review]
Looks good. Thanks for contributing.

File: src/sagemaker/deserializers.py (hunk 1 above)
Comment (athewsey) [reply src/sagemaker/deserializers.py:79]: But in the current implementation, `ACCEPT` property always returns a tuple, right? It's the`.accept` instance variable that could be str or Tuple[str]?

**Review by ajaykarpur — APPROVED** [review]

**Pull request merged**